cli/azd/internal/useragent.go

```diff
@@ -16,95 +16,30 @@ const (
-	githubActionsIdentifier string
-}
-
-func (userAgent *UserAgent) String() string {
-	var sb strings.Builder
-	sb.WriteString(userAgent.azDevCliIdentifier)
-	appendIdentifier(&sb, userAgent.userSpecifiedIdentifier)
-	appendIdentifier(&sb, userAgent.templateIdentifier)
-	appendIdentifier(&sb, userAgent.githubActionsIdentifier)
-
-	return sb.String()
-}
-
-func appendIdentifier(sb *strings.Builder, identifier string) {
-	if identifier != "" {
-		sb.WriteString(" " + identifier)
+// UserAgent() creates the user agent string for azd.
+//
+// Examples:
+//   - azdev/1.0.0 (Go 1.18; windows/amd64)
+//   - azdev/1.0.0 (Go 1.18; windows/amd64) azd-caller/1.0.0
+func UserAgent() string {
+	sb := strings.Builder{}
+	sb.WriteString(fmt.Sprintf("azdev/%s", VersionInfo().Version.String()))
+	sb.WriteString(" ")
+	sb.WriteString(runtimeInfo())
+	callerAgent := os.Getenv(AzdUserAgentEnvVar)
+	if callerAgent != "" {
+		sb.WriteString(" ")
+		sb.WriteString(callerAgent)
 	}
-}
 
-func makeUserAgent(template string) UserAgent {
-	userAgent := UserAgent{}
-	userAgent.azDevCliIdentifier = getAzDevCliIdentifier()
-	userAgent.userSpecifiedIdentifier = GetCallerUserAgent()
-	userAgent.githubActionsIdentifier = getGithubActionsIdentifier()
-	userAgent.templateIdentifier = formatTemplateIdentifier(template)
-
-	return userAgent
-}
-
-// MakeUserAgentString creates a user agent string that contains all necessary product identifiers, in increasing order:
-// - The Azure Developer CLI version, formatted as `azdev/<version>`
-// - The user specified identifier, set from `AZURE_DEV_USER_AGENT` environment variable
-// - The identifier for the template used, if applicable
-// - The identifier for GitHub Actions, if applicable
-// Examples (see test `TestUserAgentStringScenarios` for all scenarios ):
-// - `azdev/1.0.0 (Go 1.18; windows/amd64)`
-// - `azdev/1.0.0 (Go 1.18; windows/amd64) Custom-foo/1.0.0 azdtempl/my-template@1.0.0 GhActions`
-func MakeUserAgentString(template string) string {
-	userAgent := makeUserAgent(template)
-
-	return userAgent.String()
-}
-
-func getAzDevCliIdentifier() string {
-	return fmt.Sprintf("%s/%s %s", azDevProductIdentifierKey, VersionInfo().Version.String(), getPlatformInfo())
-}
-
-func getPlatformInfo() string {
-	return fmt.Sprintf("(Go %s; %s/%s)", runtime.Version(), runtime.GOOS, runtime.GOARCH)
-}
-
-// GetCallerUserAgent returns the user agent calling into azd.
-func GetCallerUserAgent() string {
-	return os.Getenv(AzdUserAgentEnvVar)
-}
-
-func getGithubActionsIdentifier() string {
-	// `GITHUB_ACTIONS` must be set to 'true' if running in GitHub Actions,
-	// see https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables
-	if isRunningInGithubActions := os.Getenv(githubActionsEnvironmentVariableName); isRunningInGithubActions == "true" {
-		return githubActionsProductIdentifierKey
+	if strings.ToLower(os.Getenv("GITHUB_ACTIONS")) == "true" {
```

Comment:
Yeah. I left it here as a marker since I have a suspicion that GitHub Actions are underrepresented in our telemetry usage data, because of the async upload model. The container shuts down pretty quickly on GitHub, and we lose usage data as a result. Don't have time to perform the analysis and make a reasonable suggestion on next course of action. Supposedly with a CI pipeline task, we could either perform upload in parallel, or a blocking upload if we favor the usage data more.